include/engine/datafacade/shared_datafacade.hpp

```diff
@@ -684,16 +684,10 @@ class SharedDataFacade final : public BaseDataFacade
 
     virtual std::string GetDatasourceName(const uint8_t datasource_name_id) const override final
     {
-        std::string result;
-
-        if (m_datasource_name_offsets.empty() ||
-            datasource_name_id > m_datasource_name_offsets.size())
-        {
-            if (datasource_name_id == 0)
-                return "lua profile";
-            return "UNKNOWN";
-        }
+        BOOST_ASSERT(m_datasource_name_offsets.size() >= 1);
+        BOOST_ASSERT(m_datasource_name_offsets.size() > datasource_name_id);
 
+        std::string result;
```

Comment:
looking at the code, the same applies to the for_each a few lines above --- maybe check the code you touched for this